### ISSUE TITLE
Android cannot use voice communication

### ISSUE THREAD

**Issue by stephen-cusi** [FWGS/xash3d-fwgs#1003]
I downloaded the latest voice build, but the voice communication function does not work, and there is no voice permission application.

**Comment by a1batross** [comment]
Not a bug. Android don't implement voice recording at this time.

**Comment by stephen-cusi** [comment]
branch：voice

**Comment by a1batross** [comment]
Don't use voice branch anymore too. 

All voice is now merged to master.

**Comment by stephen-cusi** [comment]
> 不是错误。Android 目前不支持录音。

When will Android recording be supported?

**Comment by a1batross** [comment]
When somebody will make it.

**Comment by stephen-cusi** [comment]
> When somebody will make it.

I just spent a little time adding Android recording rights😅

**Comment by a1batross** [comment]
@stephen-cusi well, just adding voice permissions won't do it.

Android code in engine/platform/android/, in particular s_opensles.c, need to open audio capture device, and connect it to engine's voice code.

We have plans to migrate back to SDL2 on Android, so when it will be done, Android will get its voice automatically. Well actually it depends on SDL2 but at least we have working audio capture with it on other systems.

**Comment by stephen-cusi** [comment]
> @stephen-cusi好吧，仅仅添加语音权限是不行的。
> 
> engine/platform/android/中的Android代码，特别是s_opensles.c，需要打开音频采集设备，并连接到引擎的语音代码。
> 
> 我们计划在 Android 上迁移回 SDL2，所以当它完成时，Android 将自动获得它的声音。实际上它依赖于 SDL2，但至少我们在其他系统上可以使用它进行音频捕获。

The principle is the same as the source engine that nill ported to the Android platform, isn't it?

> @stephen-cusi好吧，仅仅添加语音权限是不行的。
> 
> engine/platform/android/中的Android代码，特别是s_opensles.c，需要打开音频采集设备，并连接到引擎的语音代码。
> 
> 我们计划在 Android 上迁移回 SDL2，所以当它完成时，Android 将自动获得它的声音。实际上它依赖于 SDL2，但至少我们在其他系统上可以使用它进行音频捕获。



> @stephen-cusi well, just adding voice permissions won't do it.
> 
> Android code in engine/platform/android/, in particular s_opensles.c, need to open audio capture device, and connect it to engine's voice code.
> 
> We have plans to migrate back to SDL2 on Android, so when it will be done, Android will get its voice automatically. Well actually it depends on SDL2 but at least we have working audio capture with it on other systems.

The principle is the same as the source engine that nill ported to the Android platform, isn't it?

**Comment by stephen-cusi** [comment]
> @stephen-cusi好吧，仅仅添加语音权限是不行的。
> 
> engine/platform/android/中的Android代码，特别是s_opensles.c，需要打开音频采集设备，并连接到引擎的语音代码。
> 
> 我们计划在 Android 上迁移回 SDL2，所以当它完成时，Android 将自动获得它的声音。实际上它依赖于 SDL2，但至少我们在其他系统上可以使用它进行音频捕获。

The principle is the same as the source engine that Nillerusr ported to the Android platform, isn't it?

**Comment by a1batross** [comment]
@stephen-cusi I don't know what he did in his port, I don't look into leaked sources.

**Comment by magisk666** [comment]
> 

Does Android support voice now?

**Comment by a1batross** [comment]
Yes
